### PR TITLE
compile gpiochips as cpp

### PR DESCRIPTION
**Pull request by mck1117**
Trying to get all consumers of generated config structures to compile as C++, because then we can do lovely things like encode any scaling factor in the type, instead of manually specifying it everywhere.

### CODE REVIEW

**Comment by rusefillc** [comment]
@dron0gus you are being promoted into a C++ developer!

**Comment by mck1117** [comment]
@dron0gus welcome to the land of actual type safety!

**Comment by mck1117** [comment]
related #1753

**Review by dron0gus — APPROVED** [review]
With my low C++ skils looks ok...

**Comment by dron0gus** [comment]
> @dron0gus you are being promoted into a C++ developer!

You cant teach an old dog new tricks. :)

**Comment by mck1117** [comment]
> You cant teach an old dog new tricks. :)

perfectly OK to just use C++ as "C with stricter type system"

File: firmware/config/boards/subaru_eg33/board_configuration.cpp [hunk 1]
```diff
@@ -262,3 +263,97 @@ void setBoardDefaultConfiguration(void) {
...
+		[3] = {.port = GPIOE, .pad = 6},	/* INJ 8 */
+		/* ignition pre-dirvers */
+		[4] = {.port = GPIOC, .pad = 14},	/* IGN 2 */
+		[5] = {.port = GPIOC, .pad = 13},	/* IGN 4 */
+		[6] = {.port = GPIOC, .pad = 15},	/* IGN 6 */
+		[7] = {.port = GPIOI, .pad = 9},	/* IGN 8 */
+	},
+	/* en shared between two chips */
+	// TODO: dron0gus, what to do here?
+	.en = {.port = GPIOI, .pad = 7}
```
Comment (mck1117) [inline firmware/config/boards/subaru_eg33/board_configuration.cpp:335]: @dron0gus what to do about this one? C++ requires that all fields are initialized, so should we explicitly init this as no-pin or something?

**Comment by mck1117** [comment]
everybody's favorite proteus analog fail, but everything works!

**Pull request merged**